python/tk_framework_desktopserver/shotgun/scripts/execute_command.py

```diff
@@ -13,10 +13,22 @@
 import os
 
 def execute(config, project, name, entities, base_configuration, engine_name):
+    """
+    Executes an engine command in the desired environment.
+
+    :param dict config: The pipeline configuration entity.
+    :param dict project: The project entity.
+    :param list entities: The list of entities selected in the web UI when the
+        command action was triggered.
+    :param str base_configuration: The desired base pipeline configuration's
+        uri.
+    :param str engine_name: The name of the engine to bootstrap into. This
+        is most likely going to be "tk-shotgun"
+    """
+    # The local import of sgtk ensures that it occurs after sys.path is set
+    # to what the server sent over.
     import sgtk
```

Comment:
You are using sgtk to create the boostrapper and the using it reusing it after the bootstrap. I don't think this is actually causing issues in your code, but might I suggest moving everything from the `import sgtk` to `toolkit_mgr.bootstrap_engine` to another method so that the sgtk used to bootstrap is scoped to this method only?

Then you can `import sgtk`after the bootstrap happened and you are guaranteed you are using the new code.